python/ray/data/_internal/datasource/parquet_datasource.py

```diff
@@ -633,3 +650,17 @@ def sample_fragments(
     sample_bar.close()
 
     return sample_infos
+
+
+def _add_partitions_to_table(table, partitions):
+    import pyarrow as pa
+
+    for field, value in partitions.items():
+        column = pa.array([value] * len(table))
+        field_index = table.schema.get_field_index(field)
+        if field_index != -1:
+            table = table.set_column(field_index, field, column)
+        else:
+            table = table.append_column(field, column)
```

Comment:
Same problem that i've called out in [here](https://github.com/ray-project/ray/pull/47551/files#r1753122517) -- partition schema needs to be inferred once for the whole table, not per partition

Reply:
Discussed offline -- we'll default to string if the type isn't explicitly specified by the user